app/code/community/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateController.php

```diff
@@ -77,9 +77,46 @@ public function loadBlockAction()
      */
     public function saveAction()
     {
+        // some versions of Magento store the shipping method at the top level of the $_POST array
+        if ($this->getRequest()->getPost('shipping_method')) {
+            $_POST['order']['shipping_method'] = $this->getRequest()->getPost('shipping_method');
+        }
+
+        // We must assure that Magento knows to recalculate the shipping
+        $_POST['collect_shipping_rates'] = 1;
+
+        /////////////////////////////////////////////////////////////////////////////
+        // If there is no bolt reference, then it indicates this is another payment
+        // method.  In this case, we differ to Magento to handle this
+        /////////////////////////////////////////////////////////////////////////////
         $boltReference = $this->getRequest()->getPost('bolt_reference');
+
+        if (!$boltReference) {
```

Comment:
Why not bring this condition to the beginning of the function? Then if this is another payment method we just return immediately.

Reply:
+1

Reply:
@dumega @daisy1754 
Because the things before this are relevant to all shipping method.